inst/rmarkdown/templates/elsevier/resources/template.tex

```diff
@@ -194,20 +196,29 @@
 
   \title{$title$}
   $for(author)$
-  \author[$author.affiliation$]{$author.name$$if(author.footnote)$\corref{$author.footnote$}$endif$}
+  \author[$author.affiliation$]{$author.name$$if(author.footnote)$%
+  $if(author.correspondingauthor)$\corref{cor1}$endif$%
+  \fnref{$author.footnote$}$endif$}
   $if(author.email)$ \ead{$author.email$} $endif$
   $endfor$
   $for(address)$
-  \address[$address.code$]{$address.address$}
+  \affiliation[$address.code$]{$address.address$}
   $endfor$
+  \cortext[cor1]{Corresponding author}
   $for(footnote)$
-  \cortext[$footnote.code$]{$footnote.text$}
+  \fntext[$footnote.code$]{$footnote.text$}
   $endfor$
 
   \begin{abstract}
   $abstract$
   \end{abstract}
-  $if(keywords)$ \begin{keyword} $for(keywords)$$keywords$$sep$ $endfor$\end{keyword}$endif$
+  $if(keywords)$
+  \begin{keyword}
+    $for(keywords/allbutlast)$$keywords$ \sep $endfor$
+    $for(keywords/last)$$keywords$$endfor$
+  \end{keyword}
+  $endif$
```

Comment:
Yes, that's the only place that requires pandoc 2.10. I can't think of another way to do it without the user explicitly adding the \sep separators which is a bit ugly.

Reply:
yes this is cleaner, but it brings a pretty recent version of Pandoc as a requirement for this template. 
You have concerns about `elsarticle.cls` being outdated on some Linux distribution and that user won't update. Shouldn't we have the same concerns regarding Pandoc ? 

If we consider that **rticles** users are mainly RStudio users,  a pandoc version > 2.10 is available since last release. And next release with upgraded version should be soon. So we can set the minimum requirement and see if someone complain. 

Other option would be indeed to have a comment in YAML header to ask user to add `\sep` at the end of the last keyword, or do that ourself in the format by modifying the metadata before doing conversion with Pandoc.